sdk/auth/autorest/auth.go

```diff
@@ -0,0 +1,93 @@
+package autorest
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/hashicorp/go-azure-sdk/sdk/auth"
+)
+
+func AutorestAuthorizer(authorizer auth.Authorizer) *Authorizer {
+	return &Authorizer{Authorizer: authorizer}
+}
+
+type Authorizer struct {
+	auth.Authorizer
+}
+
+// WithAuthorization implements the autorest.Authorizer interface
+func (c *Authorizer) WithAuthorization() autorest.PrepareDecorator {
+	return func(p autorest.Preparer) autorest.Preparer {
+		return autorest.PreparerFunc(func(req *http.Request) (*http.Request, error) {
+			ctx := req.Context()
+			var err error
+			req, err = p.Prepare(req)
+			if err == nil {
+				token, err := c.Token(ctx, req)
+				if err != nil {
+					return nil, err
+				}
+
+				req, err = autorest.Prepare(req, autorest.WithHeader("Authorization", fmt.Sprintf("Bearer %s", token.AccessToken)))
+				if err != nil {
+					return req, err
+				}
+
+				auxTokens, err := c.AuxiliaryTokens(ctx, req)
+				if err != nil {
+					return req, err
+				}
+
+				auxTokenList := make([]string, 0)
+				for _, a := range auxTokens {
+					if a != nil && a.AccessToken != "" {
+						auxTokenList = append(auxTokenList, fmt.Sprintf("%s %s", a.TokenType, a.AccessToken))
+					}
+				}
+
+				return autorest.Prepare(req, autorest.WithHeader("x-ms-authorization-auxiliary", strings.Join(auxTokenList, ", ")))
```

Comment:
does calling this with an empty slice/value cause issues? I'd have thought we shouldn't be setting this unless there's multiple tokens present?

Reply:
This seems to be fine with an empty slice... but we get an empty header which seems messy; will fix 👍